jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/StandardDescriptorProcessor.java

```diff
@@ -1028,69 +1038,134 @@ public void addWelcomeFiles(WebAppContext context, XmlParser.Node node, Descript
         }
     }
 
+    /**
+     * Check if a path is eligible to be mapped to the servlet.
+     * A path can be (re)mapped iff:
+     * <ul>
+     *     <li>it is not already mapped</li>
+     *     <li>it is already mapped by webdefault.xml</li>
+     *     <li>it is already mapped by the embedded api, and the descriptor is not webdefault.xml</li>
+     * </ul>
+     * The effect of the above conditions is to produce the following precedence hierarchy:
+     * <ol>
+     *     <li>jetty-override.xml (OverrideDescriptor)</li>
+     *     <li>web.xml (WebDescriptor)</li>
+     *     <li>web-fragment.xml (FragmentDescriptor)</li>
+     *     <li>embedded api</li>
+     *     <li>webdefault.xml (DefaultsDescriptor)</li>
+     * </ol>
```

Comment:
This style is clearer... I.E defining a precedence hierarchy.   But the implementation below doesn't use a hierarchy?
Also it is unclear how jakarta API and annotation mappings fit into this if we are switching on origin.